pytorch_lightning/metrics/metric.py

```diff
@@ -159,38 +197,17 @@ class TensorMetric(Metric):
     Already handles DDP sync and input/output conversions.
     """
 
-    def __init__(self, name: str,
-                 reduce_group: Optional[Any] = None,
-                 reduce_op: Optional[Any] = None):
-        """
-
-        Args:
-            name: the metric's name
-            reduce_group: the process group for DDP reduces (only needed for DDP training).
-                Defaults to all processes (world)
-            reduce_op: the operation to perform during reduction within DDP (only needed for DDP training).
-                Defaults to sum.
-        """
-        super().__init__(name)
-        self.reduce_group = reduce_group
-        self.reduce_op = reduce_op
-
     @staticmethod
     def input_convert(self, data: Any):
-        return apply_to_collection(data,
-                                   (torch.Tensor, np.ndarray, numbers.Number),
-                                   convert_to_tensor,
-                                   self.dtype, self.device)
+        data = apply_to_collection(
+            data, (torch.Tensor, np.ndarray, numbers.Number), convert_to_tensor, self.dtype, self.device
+        )
+        return super(TensorMetric, self).input_convert(self, data)
 
     @staticmethod
     def output_convert(self, data: Any, output: Any):
-        return apply_to_collection(output, torch.Tensor, convert_to_tensor,
-                                   self.dtype, self.device)
-
-    @staticmethod
-    def ddp_sync(self, data: Any, output: Any):
-        return apply_to_collection(output, torch.Tensor, sync_ddp_if_available,
-                                   self.reduce_group, self.reduce_op)
+        output = apply_to_collection(output, torch.Tensor, convert_to_tensor, self.dtype, self.device)
+        return super(TensorMetric, self).output_convert(self, data, output)
 
 
 class TensorCollectionMetric(Metric):
```

Comment:
We should at some point figure out to merge `TensorMetric` with `TensorCollectionMetric` since the difference seems very subtle

Reply:
we could probably do a try on tensor aggregation and catch it for TensorCollection. And same for output convert

Reply:
Think I already merged these

Reply:
Just something to think about, it would make it a bit easier for the user to implement own stuff